src/main/scala/com/codacy/plugins/api/results/Pattern.scala

```diff
@@ -33,7 +33,8 @@ object Pattern {
                            category: Category,
                            subcategory: Option[Subcategory],
                            parameters: Option[Set[Parameter.Specification]],
-                           languages: Option[Set[Language]] = None) {
+                           languages: Option[Set[Language]] = None,
+                           enabled: Option[Boolean]) {
```

Comment:
Can you add a comment explaining that you are making this optional for the purpose of adding this value to the tools without doing a breaking change and that eventually it will be a boolean?

Reply:
Will it be a boolean eventually? If you look at the current spec we have `Option[Set[...]]` in two places already.. This means that we have two states with the same meaning: `Some(Set())` and `None`.
If the breaking change is the problem we can try to update `codacy-plugins-api` the infrastructure as last when the tools already have it 🤔

Reply:
play-json converts correctly a json object containing an extra field not in the `Specification`.

Example using current `codacy-engine-scala-seed`:
```scala
@ play.api.libs.json.Json.parse("""
  {
    "patternId": "a",
    "category": "CodeStyle",
    "level": "Error",
    "enabled": true
  }
  """)
res4: play.api.libs.json.JsValue = JsObject(
  Map(
    "patternId" -> JsString("a"),
    "category" -> JsString("CodeStyle"),
    "level" -> JsString("Error"),
    "enabled" -> JsTrue(true)
  )
)

@ res2.as[com.codacy.plugins.api.results.Pattern.Specification]
res5: results.Pattern.Specification = Specification(Id("a"), Error, CodeStyle, None, None, None)
```

We could make it non-optional right now and update tools first and codacy second, if this is what we want to do eventually.

Reply:
This will prevent us from implementing any handling of this field until we had all tools already publishing it, would it?

Reply:
I was under the impression it was only required during migration.
Ideally I think we should not have a `Troolean` since every time I saw it being used it ends up weird.
If we really need three states I think having an enumeration is prefered since the names are explicit.
What do you think?